packages/mui-base/src/Tooltip/Root/useTooltipRoot.ts

```diff
@@ -69,10 +65,10 @@ export function useTooltipRoot(params: UseTooltipRootParameters): UseTooltipRoot
 
   const { mounted, setMounted, transitionStatus } = useTransitionStatus(open, animated);
 
-  const runOnceAnimationsFinish = useAnimationsFinished(() => popupElement?.firstElementChild);
+  const runOnceAnimationsFinish = useAnimationsFinished(() => positionerElement?.firstElementChild);
```

Comment:
Could we keep reference to the popupElement and use it directly here instead of calling `firstElementChild`? Perhaps this would enable the `useAnimationsFinished to accept a ref instead of a function that returns an element?

Reply:
Yes we could pass a ref, this was just simpler with the assumption that it's always a direct child

Reply:
This assumption will break if someone writes `<Positioner render={(p) => <div {...p}><div>{p.children}</div></div>} />` or 
```
<Positioner>
  <div>
    <Popup>
```
The animation-related attributes (`data-entering`, `data-instant` are placed on the popup, so we should also look for animation on the popup)

Reply:
Changed to add `popupRef`